subsys/bluetooth/host/classic/ssp.c

```diff
@@ -443,6 +443,23 @@ void bt_hci_link_key_notify(struct net_buf *buf)
 
 	LOG_DBG("%s, link type 0x%02x", bt_addr_str(&evt->bdaddr), evt->key_type);
 
+	if (IS_ENABLED(CONFIG_BT_SMP_SC_ONLY) && (evt->key_type != BT_LK_AUTH_COMBINATION_P256)) {
+		/*
+		 * When in Secure Connections Only mode, all services
+		 * (except those allowed to have Security Mode 4, Level 0)
+		 * available on the BR/EDR physical transport require Security
+		 * Mode 4, Level 4.
+		 * Link key type should be P-256 based Secure Simple Pairing
+		 * and Secure Authentication.
+		 */
+		LOG_ERR("For SC only mode, link key type should be %d",
+			BT_LK_AUTH_COMBINATION_P256);
```

Comment:
Small improvement still: the convention when we get unexpected/invalid data from the controller is to use `LOG_WRN()` instead of `LOG_ERR()`.

Reply:
Got it. Updated.